### PR TITLE
-Author : Armine Gevorgyan - Add Calculator in Java

### PR DESCRIPTION
**Pull request by armine94**

### CODE REVIEW

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/my-app.iml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (taronqalashyan) [inline Armine_Gevorgyan/QA/CalculatorJava/my-app/my-app.iml]: add to gitignore

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        Calculator calculator = new Calculator();
+        calculator.doTheShuntingYard("((7 + 9) / 2)/0");
+    }
+
+    @Test
+    public void shouldAnswerWithTrue()
+    {
+        assertTrue( true );
+    }
+
```
Comment (taronqalashyan) [inline Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: what are you checking with this case?

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    // input  expression is empty . then expected illegalArgumentException exception with message Empty expression or null
+    @Test (expected = IllegalArgumentException.class)
+    public void emptyExpression() {
+        ExpectedException exception = ExpectedException.none();
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("Empty expression or null");
+        Calculator calculator = new Calculator();
+        calculator.doTheShuntingYard(" ");
```
Comment (taronqalashyan) [inline Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: what are you checking with this case?

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    }
+
+    // input  expression has been invalid expression . then expected illegalArgumentException exception with message Invalid Expression: 88$6
+    @Test (expected = IllegalArgumentException.class)
+    public void invalidExpression() {
+        ExpectedException exception = ExpectedException.none();
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("Invalid Expression: 88$6");
+        Calculator calculator = new Calculator();
+        calculator.doTheShuntingYard("88$6");
```
Comment (taronqalashyan) [inline Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: what are you checking with this case?

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+    // input expression has been dividing by 0 , then expected ArithmeticException exception  with message Division by 0
+    @Test (expected = ArithmeticException.class)
+    public void throwsArithmeticExeption() {
+        ExpectedException exception = ExpectedException.none();
+        exception.expect(ArithmeticException.class);
+        exception.expectMessage("Division by 0");
+        Calculator calculator = new Calculator();
+        calculator.doTheShuntingYard("((7 + 9) / 2)/0");
+    }
```
Comment (taronqalashyan) [inline Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: what are you checking with this case?

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java (hunk 3 above)
Comment (armine94) [reply Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: what doing function when expression will be empty

Comment (armine94) [reply Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: i solved that problem

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java (hunk 4 above)
Comment (armine94) [reply Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: i solved that problem

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java (hunk 5 above)
Comment (armine94) [reply Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: i solved that problem

File: Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java (hunk 2 above)
Comment (armine94) [reply Armine_Gevorgyan/QA/CalculatorJava/my-app/src/test/java/com/mycompany/app/AppTest.java]: i solved that problem

**Pull request merged**